src/app/trainingform/trainingform.component.html

```diff
@@ -0,0 +1,109 @@
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.format' | translate}}</label>
+        <mat-select class="select-group" formControlName="trainingFormat">
+          <mat-option *ngFor="let format of formats" [value]="format">
+            {{format}}
+          </mat-option>
+        </mat-select>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.language' | translate}}</label>
+        <mat-select class="select-group" formControlName="trainingLanguage">
+          <mat-option *ngFor="let language of languagies" [value]="language">
+            {{language}}
+          </mat-option>
+        </mat-select>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.requirements' | translate}}</label>
+        <textarea matInput
+          cdkTextareaAutosize
+          formControlName="trainingRequirements"
+          placeholder="{{ 'trainingform.requirements' | translate}}"
+          maxlength="1000"
+        ></textarea>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.skills' | translate}}</label>
+        <textarea matInput
+          cdkTextareaAutosize
+          formControlName="trainingSkills"
+          placeholder="{{ 'trainingform.skills' | translate}}"
+          maxlength="1000"
+        ></textarea>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.duration' | translate}}</label>
+        <mat-date-range-input [rangePicker]="picker">
+          <input matStartDate formControlName="trainingStart" placeholder="{{ 'trainingform.start-date' | translate}}">
+          <input matEndDate formControlName="trainingEnd" placeholder="{{ 'trainingform.end-date' | translate}}">
+        </mat-date-range-input>
+        <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+        <mat-date-range-picker #picker></mat-date-range-picker>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
+        <label class="star">{{ 'trainingform.registry-duration' | translate}}</label>
+        <mat-date-range-input [rangePicker]="picker1">
+          <input matStartDate formControlName="trainingRegistryStart" placeholder="{{ 'trainingform.start-date' | translate}}">
+          <input matEndDate formControlName="trainingRegistryEnd" placeholder="{{ 'trainingform.end-date' | translate}}">
+        </mat-date-range-input>
+        <mat-datepicker-toggle matSuffix [for]="picker1"></mat-datepicker-toggle>
+        <mat-date-range-picker #picker1></mat-date-range-picker>
+      </mat-form-field>
+
+      <mat-form-field appearance="standard" class="form-control">
```

Comment:
Should the publication date be assigned automatically at the time the entity is created?